### ISSUE TITLE
Items clip through the floor frequently when dropping them in dog mode

### ISSUE THREAD

**Issue by wongnata** [wongnata/PlayerDogModel_Plus#3]
Steps to reproduce:

1. Enable dog mode
2. Pick up a large object
3. Bring it into the ship
4. Crouch
5. Look down
6. Drop item

This _can_ drop the item through the floor of the ship, which is obviously problematic for when you leave the planet.

**Comment by wongnata** [comment]
I've spent a chunk of time debugging this and trying fixes, but nothing's quite made sense yet.

I've tried:

- Adjusting the drop height of the object by a flat amount
- Adjusting the target position of the object by a flat amount 
- Checking if  a) the player is in the ship, b) the object was dropped in the ship, and c) the object target position is _not_ in the ship, and then attempting to place it in the ship (using the ship bounds)
  - Tried this with variants to just update the height to be in/on the ship bounds, updating the target position to be the closest point in/on the ship bounds
- Setting the target position or y-component of the object to be the player's current position or y-component
- Adjusting the anchor points of the dog model (tried both removing the custom anchor adjustment logic _and_ tried updating it in prefix)

So far none of these seem to work as solutions. Still need to put more work into this, but right now it's not clear to me how to solve this one. It's possible one of my previous solution attempts wasn't implemented quite right, too.
